Add additional docs (FAQs and motivation) (#10)
unitaryfoundation/jeff: README.md

- Thinking beyond simple circuits will allow us to do new things and progress the field.

Compilation and optimization of such programs is crucial to support quantum hardware execution,

> however current de-facto interchange formats (such as QIR and OpenQASM) either aren't designed for
> quantum optimization — lacking the explicit structure needed for advanced high-level
> optimizations — nor tool-agnostic exchange, locking developers into particular toolchains like
> LLVM.

Reviewer: Suggest rephrasing since "quantum optimization" is ambiguous - I assume what is meant is quantum program optimization and not quantum algorithms for optimization. But also, I don't think it's the case that they aren't designed for compilation/optimization, it's more that the format is amenable only to optimization of the lowest-level (gate) instructions.

```suggestion
however current de-facto interchange formats (such as QIR and OpenQASM) either lack the explicit structure needed for advanced high-level
optimizations, or are not tool-agnostic, locking developers into particular toolchains like
LLVM.
```

Author: Good point, might make a slight modification here:

```suggestion
however current de-facto interchange formats (such as QIR and OpenQASM) either lack the explicit structure needed for advanced high-level
quantum optimizations, or are not tool-agnostic, locking developers into particular toolchains like
LLVM.
```

> ***This creates a bottleneck for collaboration and makes it difficult to develop and share

Reviewer: [I can't comment on the collapsed lines (54-56), so leaving a comment here.]  

It would be helpful to provide some more motivation as to why a serialized, in-memory representation is useful (presumably w.r.t. speed and scale?). 

EDIT: I see this is covered in the FAQs but putting it up front wouldn't hurt.

Author: I might address this in a subsequent PR!
